src/WebJobs.Script/Workers/SharedMemoryDataTransfer/ISharedMemoryManager.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Script.Workers.SharedMemoryDataTransfer
+{
+    public interface ISharedMemoryManager
+    {
+        Task<SharedMemoryMetadata> PutBytesAsync(byte[] content);
+
+        Task<SharedMemoryMetadata> PutObjectAsync(object input);
+
+        Task<byte[]> GetBytesAsync(string mapName, long offset, long count);
+
+        bool IsSupported(object input);
```

Comment:
nit: Rename to `IsDatatypeSupported` also change the input to `Type`

Reply:
This will not just be checking the type. It will check things like if min size exceeds the threshold or not (and maybe other things in the future) so keeping it generic and passing the full object.